### PR TITLE
Deploy a documentation version for each new Synapse release

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR will run a new "Deploy release-specific documentation" job whenever a push to a branch name matching `release-v*` occurs. Doing so will create/add to a folder named `vX.Y` on the `gh-pages` branch. Doing so will allow us to build up `major.minor` releases of the docs as we release Synapse.

This is especially useful for having a mechanism for keeping around documentation of old/removed features (for those running older versions of Synapse), without needing to clutter the latest copy of the docs.

After a [discussion](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$rKmkBmQle8OwTlGcoyu0BkcWXdnHW3_oap8BMgclwIY?via=matrix.org&via=vector.modular.im&via=envs.net) in #synapse-dev, we wanted to use tags to trigger the documentation deployments, which I agreed with. However, I soon realised that the bash-foo required to turn a tag of `v1.2.3rc1` into `1.2` was a lot more complex than the branch's `release-v1.2`. So, I've gone with the latter for simplicity.

In the future we'll have some UI on the website to switch between versions, but for now you can simply just change 'develop' to 'v1.2' in the URL.

You can see the workflow in action here: https://github.com/anoadragon453/synapse/runs/2850612850

### CODE REVIEW

**Review by richvdh — APPROVED** [review]
lgtm

**Comment by anoadragon453** [comment]
After a look-over, it turned out that the workflow failed to properly check for release candidates versus full releases (we were checking for the presence of `rc` in a branch name; something we never have).

The latest commit fixes this by once again checking Synapse's `__version__` str.

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -37,9 +37,15 @@ jobs:
       - name: Get the current Synapse version
         id: vars
```
Comment (clokep) [inline .github/workflows/docs.yaml:38]: I think you'll need a `shell: bash` to do the bash magic?

Comment (anoadragon453) [reply .github/workflows/docs.yaml:38]: Hmm, good point. Turns out this works in POSIX shells as well. Will edit the wording.

**Pull request merged**